### PR TITLE
docs: additions to nlu guides

### PR DESCRIPTION
**Pull request by noelweichbrodt**
### PR Checklist

Please review the [guidelines for contributing](CONTRIBUTING.md) to this repository.

- [x] I am requesting to **pull a topic/feature/bugfix branch** (right side). In other words, not _master_.
- [x] I have run `npm test` against my changes and tests pass.
- [x] I have added or edited necessary documentation, or no docs changes are needed.

### Description

update spokestack nlu docs

### CODE REVIEW

File: content/docs/Android/nlu.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ val nlu = TensorflowNLU.Builder()
     .build()
 ```
 
-The configuration properties above refer to the model itself and metadata files distributed along with it. They're stored at the root of the app's cache directory here for convenience.
+The configuration properties above refer to the three required file for the Spokestack NLU model you're using. They're stored at the root of the app's cache directory here for convenience.
```
Comment (timmywil) [inline content/docs/Android/nlu.md]:
```suggestion
The configuration properties above refer to the three required files for the Spokestack NLU model you're using. They're stored at the root of the app's cache directory here for convenience.
```

File: content/docs/Concepts/nlu.md [hunk 2]
```diff
@@ -13,15 +13,23 @@ Today's NLU systems are typically built around the intent/slot model of semantic
 
 ## Model description
 
-Spokestack's NLU models run on [TensorFlow Lite](https://www.tensorflow.org/lite) and are built on top of Google's [BERT](<https://en.wikipedia.org/wiki/BERT_(language_model)>) language model, using a pretrained version of it to leverage information learned from billions of words.
+Every Spokestack NLU model consists of three files: a Wordpiece vocabulary text file, a TensorFlow model binary file, and a metadata JSON file.
```
Comment (space-pope) [inline content/docs/Concepts/nlu.md]:
```suggestion
Every Spokestack NLU model consists of three files: a Wordpiece vocabulary (text), a TensorFlow Lite model (binary), and metadata (JSON).
```

File: content/docs/Concepts/nlu.md [hunk 3]
```diff
@@ -13,15 +13,23 @@ Today's NLU systems are typically built around the intent/slot model of semantic
...
+### Wordpiece
 
-The sequence of integers representing the user utterance is then padded with a special separator token and trailing 0s to bring the sequence to a predetermined length. That padded sequence is the model's input (shaped `[1, sequence_length]`), and the output is two tensors representing posterior probabilities for intent classification and tag classification for each token (`[1, num_intents]` and `[1, num_tags, sequence_length]`, respectively).
+Each utterance is processed using Wordpiece tokenization, a variant of [Byte pair encoding](https://en.wikipedia.org/wiki/Byte_pair_encoding) originally described in [this paper](https://static.googleusercontent.com/media/research.google.com/ja//pubs/archive/37842.pdf), to divide user utterances into tokens. The wordpiece vocabulary is pretrained and distributed along with the model; the tokens produced by the tokenizer are encoded into a sequence of integer IDs according to this vocabulary.
```
Comment (space-pope) [inline content/docs/Concepts/nlu.md]:
```suggestion
Each input string is processed using Wordpiece tokenization, a variant of [Byte pair encoding](https://en.wikipedia.org/wiki/Byte_pair_encoding) originally described in [this paper](https://static.googleusercontent.com/media/research.google.com/ja//pubs/archive/37842.pdf), to divide user utterances into tokens. Once the input has been tokenized according to the pretrained vocabulary, the tokens are encoded into a sequence of integer IDs, each ID corresponding to that token's position in the vocabulary file.
```

File: content/docs/Concepts/nlu.md [hunk 4]
```diff
@@ -13,15 +13,23 @@ Today's NLU systems are typically built around the intent/slot model of semantic
...
+
+### Tensorflow
+
+Spokestack's NLU model runs on [TensorFlow Lite](https://www.tensorflow.org/lite) and is built on top of Google's [BERT](<https://en.wikipedia.org/wiki/BERT_(language_model)>) language model, using a pretrained version of it to leverage information learned from billions of words. The sequence of integers representing the user utterance is then padded with a special separator token and trailing 0s to bring the sequence to a predetermined length. That padded sequence is the model's input (shaped `[1, sequence_length]`), and the output is two tensors representing posterior probabilities for intent classification and tag classification for each token (`[1, num_intents]` and `[1, num_tags, sequence_length]`, respectively).
```
Comment (space-pope) [inline content/docs/Concepts/nlu.md]:
```suggestion
Spokestack's NLU model runs on [TensorFlow Lite](https://www.tensorflow.org/lite) and is built on top of Google's [BERT](<https://en.wikipedia.org/wiki/BERT_(language_model)>) language model, using a pretrained version of it to leverage information learned from billions of words. 

At inference time, the sequence of token IDs described in the previous section is padded with a special separator ID and trailing 0s to bring the sequence to a predetermined length. That padded sequence is the model's input (shaped `[1, sequence_length]`), and the output is two tensors representing posterior probabilities for intent classification and tag classification for each token (`[1, num_intents]` and `[1, num_tags, sequence_length]`, respectively).
```

File: content/docs/iOS/nlu.md [hunk 5]
```diff
@@ -25,14 +25,25 @@ The following properties are configurable, but should not be changed unless a di
 
 As mentioned in the [Getting Started](getting-started) guide, initializing the Spokestack NLU is like initializing the TTS component. Both a delegate callback interface and a publisher interface are provided. The call to `classify` is the same for either interface, but the result and `DispatchQueue` that the classification runs on are handled differently.
 
+### Controller
+
+The controller initialization is straightforward--simply tell the `SpeechConfiguration` where the three required files are for the Spokestack NLU model you're using (here we assume that they're contained in the app's bundle with hardcoded names), and then initialize the controller.
```
Comment (space-pope) [inline content/docs/iOS/nlu.md]:
```suggestion
The controller initialization is straightforward--simply tell the `SpeechConfiguration` where to find the three files required for the Spokestack NLU model you're using (here we assume that they're contained in the app's bundle with hardcoded names), and then initialize the controller.
```

File: content/docs/iOS/nlu.md [hunk 6]
```diff
@@ -25,14 +25,25 @@ The following properties are configurable, but should not be changed unless a di
...
+
+class NLU {
+public let config = SpeechConfiguration()
+config.nluVocabularyPath = Bundle(for: type(of: self)).path(forResource: "vocab", ofType: "txt")
+config.nluModelPath = Bundle(for: type(of: self)).path(forResource: "nlu", ofType: "tflite")
+config.nluModelMetadataPath = Bundle(for: type(of: self)).path(forResource: "nlu", ofType: "json")
+public let nlu = try! NLUTensorflow(self, configuration: config)
+
+}
```
Comment (space-pope) [inline content/docs/iOS/nlu.md]: Suggest wrapping this in ` ```swift ... ``` ` markup. Difficult to do this in a github suggestion, as you can imagine.

File: content/docs/iOS/nlu.md [hunk 7]
```diff
@@ -29,14 +29,15 @@ As mentioned in the [Getting Started](getting-started) guide, initializing the S
...
-public let config = SpeechConfiguration()
-config.nluVocabularyPath = Bundle(for: type(of: self)).path(forResource: "vocab", ofType: "txt")
-config.nluModelPath = Bundle(for: type(of: self)).path(forResource: "nlu", ofType: "tflite")
-config.nluModelMetadataPath = Bundle(for: type(of: self)).path(forResource: "nlu", ofType: "json")
-public let nlu = try! NLUTensorflow(self, configuration: config)
-
+    public let config = SpeechConfiguration()
+    config.nluVocabularyPath = Bundle(for: type(of: self)).path(forResource: "vocab", ofType: "txt")
+    config.nluModelPath = Bundle(for: type(of: self)).path(forResource: "nlu", ofType: "tflite")
+   config.nluModelMetadataPath = Bundle(for: type(of: self)).path(forResource: "nlu", ofType: "json")
```
Comment (space-pope) [inline content/docs/iOS/nlu.md:37]: Missed a space here

**Pull request merged**